# NOTEBOOK RAPPORT SLL

L'objectif de ce notebook est de remplir le rapport du Service du Livre et de la Lecture (SLL).

# A - Site internet

## A1 - Site internet

* Site internet 
* Statut site : Site internet propre

La bibliothèque a-t-elle déposé une attestation de conformité de son site web aux exigences du Référentiel général d'accessibilité pour les administrations RGAA) ? **Non**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style("darkgrid")
import datetime

from kiblib.utils.db import DbConn
from kiblib.utils.code2libelle import Code2Libelle
from kiblib.adherent import Adherent
from kiblib.pret import Pret
from kiblib.document import Document

In [2]:
db_conn = DbConn().create_engine()

year = 2025
#date_extraction_usagers = '2023-12-27'
date_extraction_usagers = '2025-12-31'

# C - Accès et installations
## C1 - Accès et installations

|Nombre d'heures d'ouverture hebdomadaires tous publics|54|C101|
|:-----------------------------------------------------|---|----|
|Nombre de jours d'ouverture hebdomadaires|6|C115|
|Nombre de jours d'ouverture annuels|???|C102|
|La bibliothèque est-elle ouverte dimanche ?|Oui|C116|
|Nombre de places assises disponibles |232|C103 | 
| - Avec possibilité de connexions aux ordinateurs |47|C104|
|Nombre de postes informatiques publics avec accès internet|???|C106|
|La bibliothèque met-elle à disposition des usagers empêchés de lire du fait d’un handicap, des postes informatiques équipés d’aides matérielles et logicielles pour accéder aux collections ?|Non|C114|
|Nombre de photocopieurs en accès public| 1|C109|
|Nombre d'imprimantes en accès public|3|C110
|Nombre de tablettes mises à disposition du public|0|C118|
|Nombre de bibliobus|1|C112
|Liseuses mises à la dispsition du public|0|C118|
|Bornes d'écoute|1|C119|
|Nombre de consoles de jeux-vidéos|6|C121|
|Connexion wifi publique|oui|C120|

In [3]:
from datetime import datetime as dt

In [4]:
query = f"""SELECT datetime, entrees
FROM statdb.stat_entrees
WHERE YEAR(datetime)={year}"""

In [5]:
df = pd.read_sql(query,db_conn)

In [6]:
df["date"] = pd.to_datetime(df['datetime']).dt.date

In [7]:
df = df.groupby('date')['entrees'].sum().to_frame()

In [8]:
C102_NbJoursDouvertureAnnuels = len(df[(df['entrees']>50)])
print(f"C102 - Nombre de jours d'ouverture annuels : {C102_NbJoursDouvertureAnnuels}")

C102 - Nombre de jours d'ouverture annuels : 283


## C2 - Catalogue

### Notices

In [9]:
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm"""
r = pd.read_sql(query, db_conn)
r

,COUNT(bm.biblionumber)
0,232577


### Notices créées dans l'année

In [10]:
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm
    JOIN koha{year}.biblio b ON b.biblionumber = bm.biblionumber 
    WHERE YEAR(b.datecreated) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

,COUNT(bm.biblionumber)
0,8318


### Notices créées dans l'année "à la main"

In [11]:
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm
    JOIN koha{year}.biblio b ON b.biblionumber = bm.biblionumber 
    WHERE YEAR(b.datecreated) = '{year}'
    AND ExtractValue( bm.metadata, '//datafield[@tag="801"]/subfield[@code="b"]' ) = '595126101'"""
r = pd.read_sql(query, db_conn)
r

,COUNT(bm.biblionumber)
0,219


### Nombre de notices entrées par importation dans l'année

In [12]:
# Nombre de notices entrées dans l'année - nombre de notices entrées à la main
query = f"""SELECT COUNT(bm.biblionumber)
    FROM koha{year}.biblio_metadata bm
    JOIN koha{year}.biblio b ON b.biblionumber = bm.biblionumber 
    WHERE YEAR(b.datecreated) = '{year}'
    AND ExtractValue( bm.metadata, '//datafield[@tag="801"]/subfield[@code="b"]' ) != '595126101'"""
r = pd.read_sql(query, db_conn)
r

,COUNT(bm.biblionumber)
0,8099


## C3 - Locaux
|Intitulé|valeur|code SLL|
|:--|---|---|
|Surface utile nette totale (en m²)|5300|C301|
|Espaces de services Interns ( en m²)|590|C302|
|Espace de service public (en m²)|3350|C303|
|Espace manifestations et animations (en m²)|250|C304|
|Espace de magains|550|C305|
|Caféterias et surfaces détente|140|C306|
|Votre bâtiment est-il accessible au sens de la loi du 11 février 2005 ?|oui|C309|
|Précisez|Monte-personnes et ascenseurs qui permettent à chacun d'accéder à l'ensemble du bâtiment|C310|


# D - Collections

In [13]:
query = f"""SELECT
                    i.itemnumber,
                    i.barcode,
                    i.dateaccessioned,
                    i.price,
                    i.homebranch,
                    i.holdingbranch,
                    i.location,
                    i.ccode,
                    i.itemcallnumber,
                    i.notforloan,
                    i.damaged,
                    DATE(i.damaged_on),
                    i.withdrawn,
                    DATE(i.withdrawn_on),
                    i.itemlost,
                    DATE(i.itemlost_on),
                    i.onloan,
                    i.datelastborrowed,
                    i.biblionumber,
                    b.title as titre,
                    bi.publicationyear,
                    bi.itemtype,
                    i.timestamp
                FROM koha{year}.items i
                JOIN koha{year}.biblioitems bi ON i.biblionumber = bi.biblionumber
                JOIN koha{year}.biblio b ON i.biblionumber = b.biblionumber
                WHERE i.notforloan IN (-1, -2, 0, 2, -4, -3, 5)
                AND i.homebranch != 'MUS'"""
items = pd.read_sql(query, db_conn)

In [14]:
# On instancie la classe Document
# La classe Document permet d'ajouter les infos dont ont a besoin pour le SLL

exemplaires = Document(df=items, db_conn=db_conn)
exemplaires.get_doc_statdb_data()
exemplaires.get_doc_es_data()

In [15]:
exemplaires.df.columns

Index(['itemnumber', 'barcode', 'dateaccessioned', 'price', 'homebranch',
       'holdingbranch', 'location', 'ccode', 'itemcallnumber', 'notforloan',
       'damaged', 'DATE(i.damaged_on)', 'withdrawn', 'DATE(i.withdrawn_on)',
       'itemlost', 'DATE(i.itemlost_on)', 'onloan', 'datelastborrowed',
       'biblionumber', 'titre', 'publicationyear', 'itemtype', 'timestamp',
       'doc_item_id', 'doc_item_code_barre', 'doc_item_date_creation',
       'doc_item_prix', 'doc_item_site_detenteur_code',
       'doc_item_site_rattachement_code', 'doc_item_localisation_code',
       'doc_item_collection_ccode', 'doc_item_cote', 'doc_statut_code',
       'doc_statut_abime_code', 'doc_statut_desherbe_code',
       'doc_statut_perdu_code', 'doc_usage_emprunt_date',
       'doc_usage_date_dernier_pret', 'doc_biblio_id', 'doc_biblio_titre',
       'doc_biblio_annee_publication', 'doc_biblio_support_code',
       'doc_item_date_modif', 'doc_item_date_creation_annee',
       'doc_item_site_detenteur'

In [16]:
exemplaires.df

,itemnumber,barcode,dateaccessioned,price,homebranch,holdingbranch,location,ccode,itemcallnumber,notforloan,...,doc_statut,doc_statut_abime,doc_statut_desherbe,doc_statut_perdu,doc_usage_emprunt,doc_usage_date_dernier_pret_annee,doc_biblio_support,sll_public,sll_acces,sll_collection
0,1,C0001353993,2005-03-22,8.99,MED,MED,MED0A,JABEFAE,EXT E C/ALE,0,...,3 - Empruntable,Non,Non,Non,non,2017.0,Livre,enfants,accès indirect,D1 - Livres imprimés
1,3,C0000653853,2005-03-22,8.99,MED,MED,MED0A,JABEFAE,EXT E C/ALE,0,...,3 - Empruntable,Non,Non,Non restitué,emprunté,2017.0,Livre,enfants,accès indirect,D1 - Livres imprimés
2,4,C0003476991,2005-03-22,1.00,MED,MED,MED0A,JABEFAE,EXT E C/ALE,0,...,3 - Empruntable,Non,Non,Non,non,2018.0,Livre,enfants,accès indirect,D1 - Livres imprimés
3,5,C0001499529,2005-03-22,8.99,MED,MED,MED0A,JABEFAE,EXT E C/ALE,0,...,3 - Empruntable,Non,Non,Non,non,2016.0,Livre,enfants,accès indirect,D1 - Livres imprimés
4,8,C0001353935,2005-03-22,8.99,MED,MED,MED0A,JABEFAE,EXT E C/ALE,0,...,3 - Empruntable,Non,Non,Non,non,2018.0,Livre,enfants,accès indirect,D1 - Livres imprimés
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
261330,504863,None,2025-12-30,29.95,MED,MED,MED3A,ACFHSZZ,Doc./940.2 PAR,-2,...,2 - En traitement,Non,Non,Non,non,NaN,Livre,adultes,accès direct,D1 - Livres imprimés
261331,504864,C2500040372,2025-12-31,13.90,MED,MED,MED2A,JROROZZ,E R/CHI,0,...,3 - Empruntable,Non,Non,Non,non,NaN,Livre,enfants,accès direct,D1 - Livres imprimés
261332,504865,C2500040856,2025-12-31,15.90,MED,MED,MED1A,ALTROAD,ADO R/ELL,-2,...,2 - En traitement,Non,Non,Non,non,NaN,Livre,adultes,accès direct,D1 - Livres imprimés
261333,504866,C2500040857,2025-12-31,18.00,MED,MED,MED1A,ALTROAD,ADO R/RAZ,-2,...,2 - En traitement,Non,Non,Non,non,NaN,Livre,adultes,accès direct,D1 - Livres imprimés


In [17]:
exemplaires.df['sll_acq'] = 'non'
exemplaires.df.loc[exemplaires.df['doc_item_date_creation'].astype('str').str[0:4] == '2023', 'sll_acq' ] = 'oui'

In [18]:
exemplaires_sll = exemplaires.df
exemplaires_sll = exemplaires_sll[['sll_acq', 'sll_public', 'sll_acces', 'sll_collection', 'doc_biblio_id', 'doc_item_id']]

In [19]:
exemplaires_sll['sll_public'].value_counts()

adultes    188601
enfants     72734
Name: sll_public, dtype: int64

In [20]:
exemplaires_sll['sll_acq'].value_counts()

non    248934
oui     12401
Name: sll_acq, dtype: int64

In [21]:
exemplaires_sll['sll_acces'].value_counts()

accès direct      134028
accès indirect    127307
Name: sll_acces, dtype: int64

In [22]:
query = f"""SELECT
                    itemnumber,
                    barcode,
                    dateaccessioned,
                    homebranch,
                    location,
                    ccode,
                    biblionumber,
                    itemtype
                FROM statdb.stat_eliminations
                WHERE annee_mise_pilon = '{year}'"""
elim = pd.read_sql(query, db_conn)

In [23]:
eliminations = Document(df=elim, db_conn=db_conn)
eliminations.get_doc_statdb_data()
eliminations.get_doc_es_data()

In [24]:
eliminations_sll = eliminations.df
eliminations_sll = eliminations_sll[['sll_public', 'sll_acces', 'sll_collection', 'doc_biblio_id', 'doc_item_id']]

## D1 - Livres imprimés

In [25]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D1 - Livres imprimés']
data_el = eliminations_sll[eliminations_sll['sll_collection'] == 'D1 - Livres imprimés']

### Fonds
#### UM - Adultes et Enfants

In [26]:
data.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    122639
enfants     68506
Name: doc_item_id, dtype: int64

#### UM en libre-accès - Adultes et Enfants

In [27]:
data_libre_acces = data[data['sll_acces'] == 'accès direct']
data_libre_acces.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    62239
enfants    33709
Name: doc_item_id, dtype: int64

#### TITRES en libre accès - Adultes et Enfants

In [28]:
data_libre_acces = data[data['sll_acces'] == 'accès direct']
data_libre_acces.groupby(['sll_public'])['doc_biblio_id'].nunique()

sll_public
adultes    61079
enfants    29669
Name: doc_biblio_id, dtype: int64

### Acquisitions

#### Unités matérielles

In [29]:
data[data['sll_acq'] == 'oui'].groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    5240
enfants    5328
Name: doc_item_id, dtype: int64

#### Titres

In [30]:
data[(data['sll_acq'] == 'oui')
     & (data['sll_acces'] == 'accès direct')].groupby(['sll_public'])['doc_biblio_id'].nunique()

sll_public
adultes    4122
enfants    3542
Name: doc_biblio_id, dtype: int64

### Eliminations

In [31]:
data_el.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    8486
enfants    4797
Name: doc_item_id, dtype: int64

###  TEST LUCAS

In [32]:
data.groupby(['sll_public'])['doc_item_id'].nunique()

sll_public
adultes    122639
enfants     68506
Name: doc_item_id, dtype: int64

In [33]:
import numpy as np
data_fonds = data.pivot_table(index=['sll_public','sll_acces'],
                             values=['doc_item_id','doc_biblio_id'],
                             aggfunc="nunique",
                             #margins=True,
                             #margins_name='Total'
                )
#data_fonds['total'] = data_fonds['doc_biblio_id'] + data_fonds['doc_item_id']

data_fonds_concat = pd.concat([
    y.append(y.sum().rename((x, 'Sous-Total')))
    for x, y in data_fonds.groupby(level=0)]).append(data_fonds.sum().rename(('Enfants + Adultes','Total')))
data_fonds_concat

/tmp/ipykernel_20870/2752056215.py:11: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  y.append(y.sum().rename((x, 'Sous-Total')))
/tmp/ipykernel_20870/2752056215.py:11: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  y.append(y.sum().rename((x, 'Sous-Total')))
/tmp/ipykernel_20870/2752056215.py:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data_fonds_concat = pd.concat([


doc_biblio_id  doc_item_id
sll_public        sll_acces                                 
adultes           accès direct            61079        62239
                  accès indirect          57025        60400
                  Sous-Total             118104       122639
enfants           accès direct            29669        33709
                  accès indirect          25055        34797
                  Sous-Total              54724        68506
Enfants + Adultes Total                  172828       191145

## D2 - Publications en séries en cours imprmimées
> Demander à Stéphanie

In [34]:
query = f"""SELECT s.biblionumber,
s.startdate AS "debut_abonnement",
s.enddate AS "fin_abonnement",
b.title AS "titre", 
s.callnumber AS "cote",
s.location AS "localisation"
FROM koha_prod.subscription s 
LEFT JOIN biblio b on b.biblionumber = s.biblionumber
WHERE s.enddate >= '{year}-12-31'"""

In [35]:
abonnements_valides = pd.read_sql(query,db_conn)

In [36]:
loc_jeunesse = ['MED2A','MED2C']

In [37]:
abonnements_valides.loc[abonnements_valides['localisation'].isin(loc_jeunesse),
                        ['D2 - Abonnements']] = 'Jeunesse'

abonnements_valides.loc[~abonnements_valides['localisation'].isin(loc_jeunesse),
                        ['D2 - Abonnements']] = 'Adultes'

abonnements_valides['nb'] = 1

In [38]:
abonnements_valides.columns

Index(['biblionumber', 'debut_abonnement', 'fin_abonnement', 'titre', 'cote',
       'localisation', 'D2 - Abonnements', 'nb'],
      dtype='object')

In [39]:
abonnements_valides.pivot_table(index='D2 - Abonnements',
                                values=['nb','titre'],
                                aggfunc={
                                    'nb':sum,
                                    'titre':'nunique'
                                }
                               )

,nb,titre
D2 - Abonnements,,
Adultes,312,220
Jeunesse,46,35


## D3 - Autres documents

> PROPAL LUCAS

In [40]:
exemplaires_sll[exemplaires_sll['sll_collection'].str.contains('D3')].pivot_table(index='sll_collection',
                                                                   values=['doc_item_id','sll_acq'],
                                                                   aggfunc={
                                                                       'doc_item_id':'nunique',
                                                                       'sll_acq':lambda x: (x=='oui').sum()
                                                                   }
                                                                  ).rename(columns={'doc_item_id':'Fonds','sll_acq':'Acquisitions'})

,Fonds,Acquisitions
sll_collection,,
D3 - Autres documents,143,44
D3 - Documents graphiques,95,1
D3 – Documents cartographiques,81,0
D3 – Musique imprimée,3016,74


> FIN PROPAL LUCAS

###  Documents cartographiques

In [41]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 – Documents cartographiques']

In [42]:
data['doc_item_id'].nunique()

81

In [43]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

0

###  Musique imprimée

In [44]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 – Musique imprimée']

In [45]:
data['doc_item_id'].nunique()

3016

In [46]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

74

In [47]:
data[(data['sll_acces'] == 'accès direct')]['doc_item_id'].nunique()

1040

In [48]:
data[(data['sll_acces'] == 'accès direct')]['doc_biblio_id'].nunique()

1038

In [49]:
data[(data['sll_acces'] == 'accès direct') & (data['sll_acq'] == 'oui')]['doc_biblio_id'].nunique()

73

###  Documents graphiques

In [50]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 - Documents graphiques']

In [51]:
data['doc_item_id'].nunique()

95

In [52]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

1

###  Autres documents

In [53]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D3 - Autres documents']

In [54]:
data['doc_item_id'].nunique()

143

In [55]:
data[data['sll_acq'] == 'oui']['doc_item_id'].nunique()

44

### Eliminations

In [166]:
data_el = eliminations_sll[eliminations_sll['sll_collection'].str[:2] == 'D3']
print(f"D320 - Total élimination Autres documents : {len(data_el)}")

D320 - Total élimination Autres documents : 190


## D4 - Documents audiovisuels sur support

In [57]:
exemplaires_sll['sll_collection'].value_counts()

D1 - Livres imprimés                                                                  191145
D4 - Documents audiovisuels fonds adultes / Documents sonores : musique                42845
D4 - Documents audiovisuels fonds adultes / documents vidéo adultes                    12153
D1 - Publications en série imprimées                                                    7610
D3 – Musique imprimée                                                                   3016
D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés      1618
D4 - Documents audiovisuels fonds enfants / documents vidéo enfants                     1423
D4 - Documents audiovisuels fonds enfants / Documents sonores : musique                  680
D4 - Documents audiovisuels fonds enfants / Documents sonores : livres enregistrés       316
D4 - Total documents multimédia sur support                                              210
D3 - Autres documents                                                 

In [58]:
d4_coll = exemplaires_sll[exemplaires_sll['sll_collection'].str[:2] == 'D4']['sll_collection'].unique().tolist()

In [59]:
for c in d4_coll:
    print(c)
    print('-------------')
    data = exemplaires_sll[exemplaires_sll['sll_collection'] == c]
    data_el = eliminations_sll[eliminations_sll['sll_collection'] == c]
    print(f"fonds : {len(data)}")
    print(f"acquisitions : {len(data[data['sll_acq'] == 'oui'])}")
    print(f"eliminations : {len(data_el)}")
    
    print('-------------') 

D4 - Documents audiovisuels fonds adultes / Documents sonores : musique
-------------
fonds : 42845
acquisitions : 601
eliminations : 2221
-------------
D4 - Documents audiovisuels fonds enfants / Documents sonores : musique
-------------
fonds : 680
acquisitions : 22
eliminations : 29
-------------
D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés
-------------
fonds : 1618
acquisitions : 44
eliminations : 71
-------------
D4 - Documents audiovisuels fonds adultes / documents vidéo adultes
-------------
fonds : 12153
acquisitions : 365
eliminations : 628
-------------
D4 - Total documents multimédia sur support
-------------
fonds : 210
acquisitions : 1
eliminations : 8
-------------
D4 - Documents audiovisuels fonds enfants / Documents sonores : livres enregistrés
-------------
fonds : 316
acquisitions : 1
eliminations : 6
-------------
D4 - Documents audiovisuels fonds enfants / documents vidéo enfants
-------------
fonds : 1423
acquisitions : 92
eli

In [60]:
data = exemplaires_sll[exemplaires_sll['sll_collection'] == 'D1 - Livres imprimés']
data_el = eliminations_sll[eliminations_sll['sll_collection'] == 'D1 - Livres imprimés']

## D6 - Ressources numériques acquises
> Depuis juin 2023 les ressources ne sont plus acquises directement par la médiathèque de Roubaix, mais par la MEL.

### Bouquets de contenus
Pour chaque item, merci de sélectionner le bouquet principal si vous en avez plusieurs
 
|Intitulé|oui/non|Nom de l'offre|Code SLL
|:---|:---:|---|---|
    |Livre|non|   |D522
|Livres enregistrés|non|   |D524
|Dictionnaires, encyclopédies|non|   |D526|
|Musique|non|   |D528
|Vidéo|non|   |D530|
|Presse|non|   |D532|
|Apprentissage en ligne (adultes)|non|   |D534|
|Ludo-éducatif (jeunesse)|non|   |D536|
|Autres bouquets acquis|non|   |D538

### Ressources à l'unité

|Intitulé|oui/non|Nombre d'acquisitions|Nom du fournisseur|Code SLL|
|:---|---|---|---|---|
|Livres|oui|45|Autre|D539-540-541|
|Presse|non|   |   |   |
Applications|non|   |   |   |
Autres ressources|non|   |   |    |

## D7 - Collections adaptées

|intitulé|oui/non|Code SLL|
|:---|---|---|
|La bibliothèque propose-t-elle des collections adaptées ?|oui|D701
|Livres en gros caractères|oui|D708
|Livres lus du commerce|oui|D709
|Braille|non|D702
|Ouvrages en Daisy|non|
|Revues en synthèse vocale|non|D703
|Revues en synthèse vocale|non|D704
|Livres-DVD en langue des signes française (LSF)|non|D705
|Livres tactiles ou en relief|non|D706
|Autres collections adaptées|oui|D707

# E - Usages et usagers de la bibliothèque

## E1 - Usagers

In [61]:
query = f"""SELECT
    date_extraction,
    age as adh_age,
    geo_ville as adh_geo_ville,
    geo_roubaix_iris as adh_geo_rbx_iris_code,
    sexe as adh_sexe,
    inscription_code_carte as adh_inscription_carte_code,
    inscription_code_site as adh_inscription_site_code,
    inscription_attribut as adh_inscription_attribut_action_code,
    inscription_fidelite as adh_inscription_nb_annees_adhesion,
    nb_venues_prets_mediatheque,
    nb_venues_prets_bus
FROM statdb.stat_adherents WHERE date_extraction = '{date_extraction_usagers}'"""
inscrits = pd.read_sql(query, db_conn)

In [62]:
adh = Adherent(df=inscrits, db_conn=db_conn)
adh.get_adherent_statdb_data()
adh.get_adherent_es_data()

In [63]:
adh.df['nb'] = 1
adh.df['adh_age_sll'] = 'Non pertinent'
adh.df.loc[adh.df['adh_age_lib3'] == '0 - 14 ans', 'adh_age_sll'] = '0 - 14 ans'
adh.df.loc[adh.df['adh_age_lib3'] == '15 - 24 ans', 'adh_age_sll'] = '15 - 64 ans'
adh.df.loc[adh.df['adh_age_lib3'] == '25 - 64 ans', 'adh_age_sll'] = '15 - 64 ans'
adh.df.loc[adh.df['adh_age_lib3'] == '65 ans et plus', 'adh_age_sll'] = '65 ans et plus'
adh.df.loc[adh.df['adh_age_lib3'] == 'Inconnu', 'adh_age_sll'] = 'Inconnu'

### Inscrits actifs

In [64]:
data = adh.df

In [65]:
data.groupby(['adh_age_sll', 'adh_sexe'])['nb'].sum()

adh_age_sll     adh_sexe
0 - 14 ans                    12
                F           2601
                M           2385
                O              1
15 - 64 ans                   13
                F           5590
                M           3688
                O              6
65 ans et plus  F            531
                M            423
Non pertinent   N            317
Name: nb, dtype: int64

In [66]:
data.groupby(['adh_age_sll'])['nb'].sum()

adh_age_sll
0 - 14 ans        4999
15 - 64 ans       9297
65 ans et plus     954
Non pertinent      317
Name: nb, dtype: int64

In [67]:
data.groupby(['adh_age_lib2', 'adh_sexe'])['nb'].sum()

adh_age_lib2    adh_sexe
15 ans et plus                13
                F           6121
                M           4111
                O              6
<= 14 ans                     12
                F           2601
                M           2385
                O              1
Non pertinent   N            317
Name: nb, dtype: int64

In [68]:
data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()

10660

### dont nouveaux inscrits

In [69]:
data = adh.df[adh.df['adh_inscription_nb_annees_adhesion'] == 0]
len(data)

6097

In [70]:
data.groupby(['adh_age_sll', 'adh_sexe'])['nb'].sum()

adh_age_sll     adh_sexe
0 - 14 ans      F           1039
                M            969
                O              1
15 - 64 ans                    1
                F           2322
                M           1532
                O              3
65 ans et plus  F             94
                M             79
Non pertinent   N             57
Name: nb, dtype: int64

In [71]:
data.groupby(['adh_age_sll'])['nb'].sum()

adh_age_sll
0 - 14 ans        2009
15 - 64 ans       3858
65 ans et plus     173
Non pertinent       57
Name: nb, dtype: int64

In [72]:
data.groupby(['adh_age_lib2', 'adh_sexe'])['nb'].sum()

adh_age_lib2    adh_sexe
15 ans et plus                 1
                F           2416
                M           1611
                O              3
<= 14 ans       F           1039
                M            969
                O              1
Non pertinent   N             57
Name: nb, dtype: int64

In [73]:
data.groupby(['adh_age_lib2'])['nb'].sum()

adh_age_lib2
15 ans et plus    4031
<= 14 ans         2009
Non pertinent       57
Name: nb, dtype: int64

In [74]:
data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()

/tmp/ipykernel_20870/3053756961.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()


4196

In [75]:
data['adh_inscription_carte_personnalite'].value_counts()

Personne        6040
Collectivité      57
Name: adh_inscription_carte_personnalite, dtype: int64

### Emprunteurs actifs

In [76]:
adh.df.columns

Index(['date_extraction', 'adh_age', 'adh_geo_ville', 'adh_geo_rbx_iris_code',
       'adh_sexe', 'adh_inscription_carte_code', 'adh_inscription_site_code',
       'adh_inscription_attribut_action_code',
       'adh_inscription_nb_annees_adhesion', 'nb_venues_prets_mediatheque',
       'nb_venues_prets_bus', 'adh_inscription_carte_personnalite_code',
       'adh_age_code', 'adh_age_lib1', 'adh_age_lib2', 'adh_age_lib3',
       'adh_geo_gentilite', 'adh_geo_rbx_iris', 'adh_geo_rbx_quartier',
       'geo_rbx_secteur', 'inscription_carte', 'adh_inscription_carte_type',
       'adh_inscription_carte_gratuite', 'adh_inscription_carte_prix',
       'adh_inscription_carte_personnalite', 'adh_inscription_site',
       'adh_inscription_nb_annees_adhesion_tra',
       'adh_inscription_attribut_action', 'nb', 'adh_age_sll'],
      dtype='object')

In [77]:
adh.df['emprunteur'] = 'non'
adh.df.loc[adh.df['nb_venues_prets_bus'] > 0, 'emprunteur'] = 'oui'
adh.df.loc[adh.df['nb_venues_prets_mediatheque'] > 0, 'emprunteur'] = 'oui'

data = adh.df[adh.df['emprunteur'] == 'oui']
len(data)

9805

In [78]:
data.groupby(['adh_age_sll', 'adh_sexe'])['nb'].sum()

adh_age_sll     adh_sexe
0 - 14 ans                     9
                F           2035
                M           1754
                O              1
15 - 64 ans                    9
                F           3301
                M           1689
                O              4
65 ans et plus  F            432
                M            278
Non pertinent   N            293
Name: nb, dtype: int64

In [79]:
data.groupby(['adh_age_sll'])['nb'].sum()

adh_age_sll
0 - 14 ans        3799
15 - 64 ans       5003
65 ans et plus     710
Non pertinent      293
Name: nb, dtype: int64

In [80]:
data.groupby(['adh_age_lib2', 'adh_sexe'])['nb'].sum()

adh_age_lib2    adh_sexe
15 ans et plus                 9
                F           3733
                M           1967
                O              4
<= 14 ans                      9
                F           2035
                M           1754
                O              1
Non pertinent   N            293
Name: nb, dtype: int64

In [81]:
data.groupby(['adh_age_lib2'])['nb'].sum()

adh_age_lib2
15 ans et plus    5713
<= 14 ans         3799
Non pertinent      293
Name: nb, dtype: int64

In [82]:
data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()

/tmp/ipykernel_20870/3053756961.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  data[(data['adh_geo_ville'] == 'ROUBAIX') & (~adh.df['adh_age_sll'].isin(['Non pertinent', 'Inconnu']))]['nb'].sum()


6495

In [83]:
data['adh_inscription_carte_personnalite'].value_counts()

Personne        9512
Collectivité     293
Name: adh_inscription_carte_personnalite, dtype: int64

### Fréquentation : entrées dans l'établissement

In [84]:
query = f"""SELECT
    SUM(entrees)
FROM statdb.stat_entrees WHERE YEAR(datetime) = '{year}'"""
entrees = pd.read_sql(query, db_conn)
entrees

,SUM(entrees)
0,275628.0


## E2 - Prêts

In [85]:
query = f"""SELECT
    issuedate as pret_date,
    date_due as pret_date_retour_prevue,
    returndate as pret_date_retour_effectif,
    renewals as pret_nb_renouvellement,
    branch as pret_site_pret_code,
    age as adh_age,
    sexe as adh_sexe_code,
    ville as adh_geo_ville,
    iris as adh_geo_rbx_iris_code,
    branchcode as adh_inscription_site_code,
    categorycode as adh_inscription_carte_code,
    fidelite as adh_inscription_nb_annees_adhesion,
    itemnumber as doc_item_id,
    homebranch as doc_item_site_detenteur_code,
    location as doc_item_localisation_code,
    ccode as doc_item_collection_ccode,
    itemcallnumber as doc_item_cote,
    itemtype as doc_biblio_support_code,
    publicationyear as doc_biblio_annee_publication,
    biblionumber as doc_biblio_id,
    dateaccessioned as doc_item_date_creation
FROM statdb.stat_issues
WHERE YEAR(issuedate) = '{year}'"""
prts = pd.read_sql(query, db_conn)
len(prts)

352264

In [86]:
c2l = Code2Libelle(db_conn)
c2l.get_val()
c2l = c2l.dict_codes_lib

prets = Pret(df=prts, db_conn=db_conn, c2l=c2l)
prets.get_pret_statdb_data()
prets.get_pret_es_data()

In [87]:
prets_sll = prets.df[['sll_public', 'sll_acces', 'sll_prets_coll', 'sll_collection', 'sll_prets']]
prets_sll['nb'] = 1

/tmp/ipykernel_20870/3051755607.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  prets_sll['nb'] = 1


In [88]:
# Nb de prêts - hors Collectivités
data = prets_sll.pivot_table(index="sll_prets",
                             columns="sll_public",
                             values="nb",
                             aggfunc=sum,
                             margins=True,
                             margins_name='Total')
data

sll_public,adultes,enfants,Total
sll_prets,,,
Autres documents,1075,60,1135
E2 - Documents vidéo,16115,3735,19850
E2 - Livres,102333,200182,302515
E2 – Documents sonores : livres,1401,366,1767
E2 – Documents sonores : musique,9550,444,9994
E2 – Publications en série imprimées,11989,5014,17003
Total,142463,209801,352264


In [89]:
# Nb de prêts - Collectivités
data_coll = prets_sll[prets_sll['sll_prets_coll']=="Prêt aux collectivités"].pivot_table(index="sll_prets",
                                                                                         #columns="sll_public",
                                                                                         values="nb",
                                                                                         aggfunc=sum,
                                                                                         margins=True,
                                                                                         margins_name='Total').rename(columns={"nb":"Dont Collectivités"})
data_coll

,Dont Collectivités
sll_prets,
Autres documents,55
E2 - Documents vidéo,75
E2 - Livres,30832
E2 – Documents sonores : livres,31
E2 – Documents sonores : musique,17
E2 – Publications en série imprimées,40
Total,31050


In [90]:
data_synthese = data.merge(data_coll,
                           left_on='sll_prets',
                           right_on='sll_prets',
                           how='left'
                          )
data_synthese = data_synthese[["adultes",'enfants','Total','Dont Collectivités']]

In [91]:
lst = ["E2 - Livres",
       "E2 – Publications en série imprimées",
       "E2 – Documents sonores : musique",
       "E2 – Documents sonores : livres",
       "E2 - Documents vidéo",
       "Autres documents",
       "Total"
      ]
data_synthese.loc[lst]

,adultes,enfants,Total,Dont Collectivités
sll_prets,,,,
E2 - Livres,102333,200182,302515,30832
E2 – Publications en série imprimées,11989,5014,17003,40
E2 – Documents sonores : musique,9550,444,9994,17
E2 – Documents sonores : livres,1401,366,1767,31
E2 - Documents vidéo,16115,3735,19850,75
Autres documents,1075,60,1135,55
Total,142463,209801,352264,31050


> A la dernière ligne **Tota** il faut ajouter pour valeur le **Nb de prêts de livres numériques**. Ce chiffre est à récupérer auprès de Mathilde L.

## E3 - Autres types d'usages

### Réservations

In [92]:
query = f"""SELECT
    COUNT(*)
FROM statdb.stat_reserves WHERE YEAR(reservedate) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

,COUNT(*)
0,55532


## E5 - Sessions internet (connexions + wifi)

In [93]:
query = f"""SELECT
    COUNT(*)
FROM statdb.stat_webkiosk WHERE YEAR(heure_deb) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

,COUNT(*)
0,61801


In [94]:
query = f"""SELECT
    COUNT(*)
FROM statdb.stat_wifi WHERE YEAR(start_wifi) = '{year}'"""
r = pd.read_sql(query, db_conn)
r

,COUNT(*)
0,11739


### E502 - Nombre de visites du site dans la bibliothèque (visite interne)

In [95]:
query = f"""SELECT SUM(visites) AS 'Total visites interne'
FROM statdb.stat_web2 sw 
WHERE YEAR(`date`) = {year}
AND origine = 'interne'"""

pd.read_sql(query,db_conn)

,Total visites interne
0,13715.0


### E503 - Nombre de visite du site internet hors bibliothèque (visite externe)

In [96]:
query = f"""SELECT SUM(visites) AS 'Total visites externe'
FROM statdb.stat_web2 sw 
WHERE YEAR(`date`) = {year}
AND origine = 'externe'"""

pd.read_sql(query,db_conn)

,Total visites externe
0,188487.0


# H - Action culturelle (nouvelle version)

* [ ] Mettre à jour le script suite à la modif du tableau source
* [ ] Vérifier les étapes stats
* [ ] Créer une classe en Python pour gérer les stats sll & stats pour collègues

> ***NOTE : avant de lancer le scrip***   
    1. Récupérer la version la plus récente du fichier `tableaux_suivi_AECS`  
    2. Importer le fichier dans le répertoire `/home/kibini/kibini2/kibini/data/aecs`  
    3. Ajouter la date d'import dans kibini a la fin du nom du fichier. 
    Exemple : `tableaux_suivi_AECS_20250917`

In [97]:
from kiblib.utils.aecs import AECS

In [98]:
sll = AECS(set_sll_columns=True)

In [99]:
sll.get_SLL_datas()

In [100]:
sll.aecs_H5XX_NbActionsHorsLesMursMed

77

# H1 - Etablissement scolaire et universitaire

In [101]:
sll.aecs_H1XX_df

,Total personnes distinctes touchées
H1 - Partenariats avec des institutions,
a/ Écoles,3966.0
b/ Collèges,1041.0
c/ Lycées,262.0
d/ Supérieur,349.0
e/ Maisons de retraite,83.0
f/ Centres sociaux,645.0
g/ Centres de loisirs,12.0
h/ Services de la petite enfance,72.0
i/ Services de l'emploi,196.0


# H2 - Équipements culturels

In [102]:
sll.aecs_H2XX_df

,Total personnes distinctes touchées
Nom d'équipement,
Archives Nationales du Monde du Travail Roubaix,15.0
Autres,38.0
Cie des Vagabondes,45.0
Compagnie de l'Oiseau-Mouche,35.0
Conservatoire de Roubaix,92.0
Librairie Les Quatre Chemins,2.0


# H3 - Structures associatives

# H4 - Actions au sein de l'établissement

In [103]:
sll.aecs_H4XX_df

Nb actions                       \
sll_type_public                             a/ Enfants b/ Tout public Total   
H4 - Actions au sein de l'établissement                                       
H4 - Autres                                         45            291   248   
H4 - Clubs de lecteurs, ateliers d'écriture          2             33    29   
H4 - Concerts, projections                          20             31    45   
H4 - Conférences, rencontres, lectures              32            278   289   
H4 - Exposition                                      0             13    11   
H4 - Séances de contes                              14              7    16   
Total                                               99            539   638   

                                            Nb personnes touchées  \
sll_type_public                                        a/ Enfants   
H4 - Actions au sein de l'établissement                             
H4 - Autres                                                  1189   
H4 - Clubs de lecteurs, ateliers d'écriture                     8   
H4 - Concerts, projections                                   1194   
H4 - Conférences, rencontres, lectures                       1470   
H4 - Exposition                                                 0   
H4 - Séances de contes                                        259   
Total                                                        4120   

                                                                     
sll_type_public                             b/ Tout public    Total  
H4 - Actions au sein de l'établissement                              
H4 - Autres                                           9987  11176.0  
H4 - Clubs de lecteurs, ateliers d'écriture            374    382.0  
H4 - Concerts, projections                             937   2131.0  
H4 - Conférences, rencontres, lectures                8100   9570.0  
H4 - Exposition                                        630    630.0  
H4 - Séances de contes                                 207    466.0  
Total                                                20235  24355.0

# H5 - Actions hors de l'établissement

In [104]:
print(f"H501 - Actions hor de l'établissement : OUI")
print(f"H502 - Nb action hors les murs : {sll.aecs_H502_NbActionsHorsLesMurs}")
print(f"H503 - Population touchée : {sll.aecs_H503_PopulationTouchee}")
print(f"H504 - Nb de prêts à domicile : Demander au service de prêts à domicile")

H501 - Actions hor de l'établissement : OUI
H502 - Nb action hors les murs : 1928
H503 - Population touchée : 7546.0
H504 - Nb de prêts à domicile : Demander au service de prêts à domicile


# H6 - Actions de coopération


|Intitulé|Oui/Non?|Champ|
|:---|---|---|
|Coopération locale|Oui|H601
|- Dont dont actions de coopération intercommunale (avec des bibliothèques n’appartenant pas à votre collectivité)|Oui|H610|
|Catalogue commun|Oui|H612|
|Portail documentaire commun|Non|H613|
|Politique documentaire commune|Non|H613
|Navette documentaire|Non|H614
|Carte unique de lecteur|Non|H615
|Action culturelle commune|Oui|H617
|Autre action de coopération intercommunal|Non|H616
|Coopération nationale|Oui|H604
|Nombre d'actions|4|H605
|Coopération internationale|Non|H607

# H7 Actions et services à destination de publics à besoins spécifiques

In [105]:
sll.aecs_H7XX_df

,Nb actions,Total personnes distinctes touchées
H7 - Actions et services à destination de publics à besoins spécifiques,,
a/ Personnes âgées,8,63.0
b/ Personnes en situation de hancidap,1,16.0
c/ Jeunes,10,197.0
d/ Petite enfance,101,2062.0
g/ Population non-francophone,23,348.0
h/ Population en situation d'insertion sociale,3,27.0


# H - Action culturelle (ancienne version

In [106]:
import pandas as pd

In [107]:
#ac = pd.read_excel("/home/kibini/AC_synthese_2023-2024.xlsx")
ac = pd.read_excel("../data/aecs/tableaux_suivi_AECS_20250917.xlsx",sheet_name="Mathilde M")

In [108]:
#ac = ac[ac['Date'].dt.year==2024]

ac = ac[ac['Date début action'].dt.year==2024]

In [109]:
# SLL Types d'action
ac.loc[ac["Type d'action"]=="Exposition",["sll_type_action"]] = "H4 - Exposition"

ac.loc[ac["Type d'action"].isin(['Conférence','Rencontre','Lecture']),['sll_type_action']] = "H4 - Conférences, rencontres, lectures"

ac.loc[ac["Type d'action"].isin(['Concert','Projection']),['sll_type_action']] = "H4 - Concerts, projections"

ac.loc[ac["Type d'action"]=="Séance de contes",["sll_type_action"]] = "H4 - Séances de contes"

ac.loc[ac["Type d'action"].isin(["Club lecture","Atelier d'écriture"]),['sll_type_action']] = "H4 - Clubs de lecteurs, ateliers d'écriture"

#ac.loc[ac["Type d'action"].isin(["Journée festive","Salon du livre","Festival"]),['sll-type_action']] = "H4 - Fêtes, salons du livre, festivals" 
#ac.loc[ac["Evénement"].notna(),["sll_type_action"]] = "H4 - Fêtes, salons du livre, festivals"


ac.loc[~ac["sll_type_action"].isin(["H4 - Exposition",
                                    "H4 - Conférences, rencontres, lectures",
                                    "H4 - Concerts, projections",
                                    "H4 - Séances de contes",
                                    "H4 - Clubs de lecteurs, ateliers d'écriture",
                                    #"H4 - Fêtes, salons du livre, festivals"
                                   ]),
       ["sll_type_action"]] = "H4 - Autres"


ac.groupby("Type d'action").size()

Type d'action
Atelier                  55
Atelier d'écriture        6
Atelier informatique      9
Autre                     2
Club lecture              7
Concert                  12
Conférence               25
Diffusion                 1
Exposition                3
Jeux                     10
Jeux-vidéo               26
Lecture                 131
Projection                6
Rencontre                11
Spectacle                16
Séance de contes          7
Vente                     3
dtype: int64

> La source de donnée utilisée pour réaliser ces stats est amené à évoluer. Il faudra bien penser à finaliser ce script une fois que les noms de colonnes du fichier Excel Action Culturel seront définitivement fixés

In [110]:
# SLL Types de Type de publics

ac.loc[ac["Type de public"].isin(["Adolescents",
                                  "Adultes",
                                  "Familles",
                                  "Tout Type de public"]),"sll_type_public"] = "a/ Tout Type de public"

ac.loc[ac['Type de public'].isin(["Enfants",
                                  "Petite enfance"]),"sll_type_public"] = "b/ Enfants"

In [111]:
ac['Nb actions'] = 1

In [112]:
# import du fichier de suivi des actions AECS
sheets2keep = ['Bruno',
               'Céline',
               'Esther',
               'Gwen',
               'Hélène',
               'Inès',
               'Laetitia C',
               'Laetitia D',
               'Mathilde M',
               'Pascale',
               'Zina',
              ]

aecs = pd.read_excel("../data/aecs/tableaux_suivi_AECS_20250917.xlsx",sheet_name=sheets2keep)

aecs

{'Bruno':    Date début action Date fin action          Champ d'action  \
 0         2024-10-04      2023-10-18   dont Livre et lecture   
 1         2024-10-13      2023-05-24               Education   
 2         2024-10-13      2023-05-24               Education   
 3         2024-10-20      2023-05-23               Education   
 4         2024-10-20      2023-05-23               Education   
 5         2024-11-09      2023-11-09               Education   
 6         2024-11-17      2023-11-17   dont Livre et lecture   
 7         2024-12-07      2023-12-15               Education   
 8         2024-12-14      2024-02-25   dont Livre et lecture   
 9         2024-11-08      2024-12-13   dont Livre et lecture   
 10        2024-01-10      2024-02-21   dont Livre et lecture   
 11        2023-12-11      2024-02-02               Education   
 12        2024-02-01      2024-03-29   dont Livre et lecture   
 13        2024-02-16      2024-03-15               Education   
 14        2024-

In [113]:
#"\\ntrbx.local\partages\mediatheque\partage_media\Evaluation\Cartographie et éval actions hors-les-murs\Nouveaux tableaux de suivi AECS\tableaux_suivi_AECS.xlsx"

In [114]:
aecs = pd.concat(aecs)

In [115]:
#effectifs_scolaires = pd.read_excel("/home/kibini/formulaire_AECS.xlsx",sheet_name="Effectifs")

effectifs_scolaires = pd.read_excel("../data/aecs/tableaux_suivi_AECS_20250917.xlsx",
                                   sheet_name="Effectifs")

effectifs_scolaires

,Nom,Total Maternelle,Total Elementaire,Total Ecole
0,Albert Camus,215.0,324.0,539
1,Albert Samain,150.0,0.0,150
2,Alphonse Daudet,24.0,0.0,24
3,Anatole France,112.0,240.0,352
4,Blaise Pascal,57.0,104.0,161
5,Boileau / Pasteur,156.0,295.0,451
6,Buffon,127.0,219.0,346
7,Charles De Foucauld,NaN,NaN,0
8,Charles Perrault,32.0,NaN,32
9,Condorcet,NaN,137.0,137


In [116]:
effectifs_scolaires.loc[effectifs_scolaires['Nom']=='Saint Louis']["Total Elementaire"].values[0]

148.0

In [117]:
aecs.columns

Index(['Date début action', 'Date fin action', 'Champ d'action',
       'Type d'action', 'Nom action ou projet', 'Lieu',
       'Espace (si hors les murs et pas dans la structure partenaire',
       'Type de public', 'Niveau si école', 'En partenariat avec',
       'Type de structure', 'Nom d'équipement', 'Nom du porteur de l'action',
       'Nb de classes touchées si public scolaire', 'Nb enfants touchés',
       'Nb adultes touchés', 'Nb d'accueil ou séance',
       'Total personnes distinctes touchée', 'Volume personnes touchées',
       'Notes', 'Total personnes distinctes touchées', 'Code de malle',
       'Titre de malle', 'Nom emprunteur', 'Nombre de docs', 'Evenement',
       'Espace (si dans les murs)', 'Partenariat (ne plus utiliser à terme)',
       'Nb personnes touchées', 'Jauge'],
      dtype='object')

### Calcul des actions collectivités

In [118]:
# Remise à plat du df
aecs = aecs.reset_index()
aecs.head(5)

,level_0,level_1,Date début action,Date fin action,Champ d'action,Type d'action,Nom action ou projet,Lieu,Espace (si hors les murs et pas dans la structure partenaire,Type de public,...,Total personnes distinctes touchées,Code de malle,Titre de malle,Nom emprunteur,Nombre de docs,Evenement,Espace (si dans les murs),Partenariat (ne plus utiliser à terme),Nb personnes touchées,Jauge
0,Bruno,0,2024-10-04 00:00:00,2023-10-18 00:00:00,dont Livre et lecture,"Projet, accompagnement de projet / accueil ave...",Turgotins passeurs d'histoire,Dans les murs,NaN,Adolescents (12-18 ans),...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Bruno,1,2024-10-13 00:00:00,2023-05-24 00:00:00,Education,Classe patrimoine,Classe patrimoine,Dans les murs,NaN,Enfant Scolaires (3-12 ans),...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Bruno,2,2024-10-13 00:00:00,2023-05-24 00:00:00,Education,Classe patrimoine,Classe patrimoine,Dans les murs,NaN,Enfant Scolaires (3-12 ans),...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,Bruno,3,2024-10-20 00:00:00,2023-05-23 00:00:00,Education,Classe patrimoine,Classe patrimoine,Dans les murs,NaN,Enfant Scolaires (3-12 ans),...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Bruno,4,2024-10-20 00:00:00,2023-05-23 00:00:00,Education,Classe patrimoine,Classe patrimoine,Dans les murs,NaN,Enfant Scolaires (3-12 ans),...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [119]:
aecs.drop(labels='level_1',axis='columns',inplace=True)

In [120]:
aecs.rename(columns={'level_0':'nom_feuille'},inplace=True)

In [121]:
# Filtre Malles Collectivités
aecs_coll = aecs[aecs["nom_feuille"]=='Laetitia D']

# Sélection des Type de structures écoles

# Liste des noms d'école
school_names = aecs_coll[aecs_coll["Type de structure"]=='Ecole']["Nom d'équipement"].unique()

for name in school_names:
    # On définit un aecs qui filtre à partir de l'itération school_name
    aecs_coll_school = aecs_coll[aecs_coll["Nom d'équipement"]==name]
    # On calcule un total par types d'actions malles
    Total_actions_BCD = len(aecs_coll_school[aecs_coll_school['Nom action ou projet']=='BCD'])
    Total_actions_MARM = len(aecs_coll_school[aecs_coll_school['Nom action ou projet']=='MARMOTHEQUE'])
    Total_emprunteurs_distincts = aecs_coll_school['Nom emprunteur'].nunique()
    
    # On insère dans le dataframe de base les colonnes du dataframe par école
    aecs_coll.loc[aecs_coll["Nom d'équipement"]==name,['Total_actions_BCD']] = Total_actions_BCD
    aecs_coll.loc[aecs_coll["Nom d'équipement"]==name,['Total_actions_MARM']] = Total_actions_MARM
    aecs_coll.loc[aecs_coll["Nom d'équipement"]==name,['Total_emprunteurs_distincts']] = Total_emprunteurs_distincts

aecs_coll

,nom_feuille,Date début action,Date fin action,Champ d'action,Type d'action,Nom action ou projet,Lieu,Espace (si hors les murs et pas dans la structure partenaire,Type de public,Niveau si école,...,Nom emprunteur,Nombre de docs,Evenement,Espace (si dans les murs),Partenariat (ne plus utiliser à terme),Nb personnes touchées,Jauge,Total_actions_BCD,Total_actions_MARM,Total_emprunteurs_distincts
362,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,HUCHETTE Juliette,28,NaN,NaN,NaN,NaN,NaN,0.0,0.0,15.0
363,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),MS,...,DRONNEAU Anne-Chrystel,14,NaN,NaN,NaN,NaN,NaN,0.0,0.0,15.0
364,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),GS,...,DEREUMAUX Camille,19,NaN,NaN,NaN,NaN,NaN,0.0,0.0,15.0
365,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,DUFOURMONT Angélique,28,NaN,NaN,NaN,NaN,NaN,0.0,0.0,15.0
366,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,ZIARNOWSKI Delphine,23,NaN,NaN,NaN,NaN,NaN,0.0,0.0,15.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2208,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,MARTEL Emilie,3,NaN,NaN,NaN,NaN,NaN,0.0,0.0,7.0
2209,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CE1,...,LEROY Sandrine,27,NaN,NaN,NaN,NaN,NaN,0.0,0.0,7.0
2210,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),PS / MS,...,VANDESTIEN Joséphine,21,NaN,NaN,NaN,NaN,NaN,0.0,0.0,5.0
2211,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),GS,...,DUPONT Marine,25,NaN,NaN,NaN,NaN,NaN,0.0,0.0,5.0


In [122]:
aecs_coll_merged = aecs_coll.merge(effectifs_scolaires,left_on="Nom d'équipement",right_on="Nom",how="left")

In [123]:
# Calcule des publics touchés

effectif_moyen_classe = 25


# Si au moins 1 action MARM : Compter l'effectif Total Maternlle
aecs_coll_merged.loc[aecs_coll_merged['Total_actions_MARM']>0,['Nb enfants touchés']] = aecs_coll_merged['Total Maternelle']

# Si au moins 1 action BCD : Compter l'effectif total Ecole
aecs_coll_merged.loc[aecs_coll_merged['Total_actions_BCD']>0,['Nb enfants touchés']] = aecs_coll_merged['Total Ecole']

# Aucune action MARM ou BCD : Estimation = Nb emprunteurs distincts x Moyenne d'une classe
aecs_coll_merged.loc[(aecs_coll_merged['Total_actions_MARM']==0)&
            (aecs_coll_merged['Total_actions_BCD']==0),['Nb enfants touchés']] = aecs_coll_merged['Total_emprunteurs_distincts'] * effectif_moyen_classe

aecs_coll_merged

,nom_feuille,Date début action,Date fin action,Champ d'action,Type d'action,Nom action ou projet,Lieu,Espace (si hors les murs et pas dans la structure partenaire,Type de public,Niveau si école,...,Partenariat (ne plus utiliser à terme),Nb personnes touchées,Jauge,Total_actions_BCD,Total_actions_MARM,Total_emprunteurs_distincts,Nom,Total Maternelle,Total Elementaire,Total Ecole
0,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,NaN,NaN,NaN,0.0,0.0,15.0,Jules Guesde,123.0,200.0,323.0
1,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),MS,...,NaN,NaN,NaN,0.0,0.0,15.0,Jules Guesde,123.0,200.0,323.0
2,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),GS,...,NaN,NaN,NaN,0.0,0.0,15.0,Jules Guesde,123.0,200.0,323.0
3,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,NaN,NaN,NaN,0.0,0.0,15.0,Jules Guesde,123.0,200.0,323.0
4,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,NaN,NaN,NaN,0.0,0.0,15.0,Jules Guesde,123.0,200.0,323.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1846,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,NaN,NaN,NaN,0.0,0.0,7.0,Léon Marlot,NaN,223.0,223.0
1847,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CE1,...,NaN,NaN,NaN,0.0,0.0,7.0,Léon Marlot,NaN,223.0,223.0
1848,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),PS / MS,...,NaN,NaN,NaN,0.0,0.0,5.0,Jules Verne,198.0,NaN,198.0
1849,Laetitia D,2025-04-25 00:00:00,2025-06-27 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),GS,...,NaN,NaN,NaN,0.0,0.0,5.0,Jules Verne,198.0,NaN,198.0


In [124]:
# Suppression des noms doublons pour les noms d'école
aecs_coll_merged.drop_duplicates(subset="Nom d'équipement",
                                 inplace=True
                                )

In [125]:
# tableau spécial collectivités
aecs_coll_merged

,nom_feuille,Date début action,Date fin action,Champ d'action,Type d'action,Nom action ou projet,Lieu,Espace (si hors les murs et pas dans la structure partenaire,Type de public,Niveau si école,...,Partenariat (ne plus utiliser à terme),Nb personnes touchées,Jauge,Total_actions_BCD,Total_actions_MARM,Total_emprunteurs_distincts,Nom,Total Maternelle,Total Elementaire,Total Ecole
0,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,NaN,NaN,NaN,0.0,0.0,15.0,Jules Guesde,123.0,200.0,323.0
6,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CE2,...,NaN,NaN,NaN,0.0,0.0,9.0,Léo Lagrange,72.0,178.0,250.0
8,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),PS,...,NaN,NaN,NaN,0.0,0.0,7.0,Pierre Brossolette,80.0,148.0,228.0
10,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CP,...,NaN,NaN,NaN,2.0,0.0,4.0,Léon Jouhaux,NaN,113.0,113.0
12,Laetitia D,2023-09-05 00:00:00,2023-10-17 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),CE2,...,NaN,NaN,NaN,2.0,4.0,18.0,Lavoisier,NaN,266.0,266.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1137,Laetitia D,2024-10-16 00:00:00,2025-04-23 00:00:00,NaN,NaN,MARMOTHEQUE,Hors les murs,NaN,Petite enfance (0-3 ans),NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1138,Laetitia D,2024-10-16 00:00:00,2025-04-23 00:00:00,NaN,NaN,MARMOTHEQUE,Hors les murs,NaN,Petite enfance (0-3 ans),NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1139,Laetitia D,2024-10-17 00:00:00,2025-06-30 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1243,Laetitia D,2024-11-07 00:00:00,2024-12-19 00:00:00,NaN,NaN,Malles,Hors les murs,NaN,Enfant Scolaires (3-12 ans),PS,...,NaN,NaN,NaN,0.0,0.0,1.0,Saint Louis,76.0,148.0,224.0


In [126]:
# Afficher le nombre de personnes touchées par école
aecs_coll_merged[aecs_coll_merged["Type de structure"]=="Ecole"][["Nom d'équipement","Nb enfants touchés"]]

,Nom d'équipement,Nb enfants touchés
0,Jules Guesde,375.0
6,Léo Lagrange,225.0
8,Pierre Brossolette,175.0
10,Léon Jouhaux,113.0
12,Lavoisier,266.0
21,Jean Macé,231.0
30,Marie Auxiliatrice,100.0
34,Michelet,300.0
44,Pierre de Ronsard niv.2,154.0
45,Montesquieu,116.0


In [127]:
nb_actions_prets_malles_2024 = aecs_coll_merged['Nb enfants touchés'].sum()

In [128]:
nb_actions_prets_malles_2024

8488.0

### Fin du calcul

In [129]:
# Rajout d'une colonne pour compter le nombre de valeurs
aecs['nb'] = 1

In [130]:
# Si le champ Nb de séance n'est pas rempli, on considère qu'il s'agit d'une occurence unique
aecs.loc[aecs["Nb d'accueil ou séance"].isna(),["Nb d'accueil ou séance"]] = 1

In [131]:
aecs["Total personnes touchées"] = aecs["Nb enfants touchés"] + aecs["Nb adultes touchés"]

## H1 - Partenariats avec des institutions

In [132]:
aecs.dtypes

nom_feuille                                                      object
Date début action                                                object
Date fin action                                                  object
Champ d'action                                                   object
Type d'action                                                    object
Nom action ou projet                                             object
Lieu                                                             object
Espace (si hors les murs et pas dans la structure partenaire     object
Type de public                                                   object
Niveau si école                                                  object
En partenariat avec                                              object
Type de structure                                                object
Nom d'équipement                                                 object
Nom du porteur de l'action                                      

In [133]:
aecs.groupby("Type de structure").size()

Type de structure
Association                         68
Centre_culturel                     11
Centre_de_loisirs                    5
Centre_social                       58
Collège                             15
Ecole                             1915
Equipement_médicosocial             13
Lycée                               12
Maison_de_retraite                   8
Service_emploi_et_formation         18
Structure_de_la_petite_enfance      83
Supérieur                           13
dtype: int64

In [134]:
# Ajout d'une colonne H1

aecs.loc[aecs['Type de structure']=="Ecole",
         ["H1 - Partenariats avec des institutions"]] = "a/ Écoles"

aecs.loc[aecs['Type de structure']=="Collège",
         ["H1 - Partenariats avec des institutions"]] = "b/ Collège"

aecs.loc[aecs['Type de structure']=="Lycée",
         ["H1 - Partenariats avec des institutions"]] = "c/ Lycée"

aecs.loc[aecs['Type de structure']=="Supérieur",
         ["H1 - Partenariats avec des institutions"]] = "d/ Supérieur"


In [135]:
# Il n'y a pour l'instant aucun partenariat avec
# Des hôpitaux 
# Des prisons

aecs.loc[aecs['Type de structure']=="Maison_de_retraite",
         ["H1 - Partenariats avec des institutions"]] = "g/ Maisons de retraite"


aecs.loc[aecs['Type de structure']=="Centre_social",
         ["H1 - Partenariats avec des institutions"]] = "h/ Centres sociaux"


aecs.loc[aecs['Type de structure']=="Centre_de_loisirs",
         ["H1 - Partenariats avec des institutions"]] = "i/ Centres de loisirs"

aecs.loc[aecs['Type de structure']=="Type de structure_de_la_petite_enfance",
         ["H1 - Partenariats avec des institutions"]] = "j/ Services de la petite enfance"

aecs.loc[aecs['Type de structure']=="Service_emploi_et_formation",
         ["H1 - Partenariats avec des institutions"]] = "k/ Services de l'emploi"

aecs.loc[aecs['Type de structure']=="Equipement_médicosocial",
         ["H1 - Partenariats avec des institutions"]] = "l/ Equipement_médicosocial"

In [136]:
aecs[aecs["nom_feuille"]!="Laetitia D"].pivot_table(index='H1 - Partenariats avec des institutions',
                                                    values=["Nb de classes touchées si public scolaire",
                                                            "Nb d'accueil ou séance",
                                                            "Total personnes touchées"],
                                                    aggfunc=sum)[["Nb de classes touchées si public scolaire",
                                                                  "Nb d'accueil ou séance",
                                                                  "Total personnes touchées"
                                                                 ]]

,Nb de classes touchées si public scolaire,Nb d'accueil ou séance,Total personnes touchées
H1 - Partenariats avec des institutions,,,
a/ Écoles,189.0,279.0,3966.0
b/ Collège,55.0,49.0,1041.0
c/ Lycée,11.0,16.0,262.0
d/ Supérieur,0.0,13.0,209.0
g/ Maisons de retraite,0.0,8.0,83.0
h/ Centres sociaux,0.0,72.0,604.0
i/ Centres de loisirs,0.0,1.0,12.0
k/ Services de l'emploi,0.0,27.0,196.0
l/ Equipement_médicosocial,0.0,10.0,60.0


> Pour la section H1 - Partenariats avec les institutions, on exclus les actions de prêts aux Collectivités

## H2 - Partenariats avec des equipements culturels 

## H3 - Partenariats avec des Type de structures associatives

In [137]:
aecs[aecs["Type de structure"]=="Association"].groupby("Champ d'action").size()

Champ d'action
Culture    26
dtype: int64

## H4 - Actions au sein de l'établissement

In [138]:
(ac.pivot_table(index='sll_type_action',
               columns='sll_type_public',
               values=['Nb actions','Nb personnes touchées'],
              aggfunc=sum,
               margins=True,
               fill_value=0,
               margins_name='Total')
 .astype(int))

Nb actions             \
sll_type_public                             a/ Tout Type de public b/ Enfants   
sll_type_action                                                                 
H4 - Autres                                                     48         13   
H4 - Clubs de lecteurs, ateliers d'écriture                     12          1   
H4 - Concerts, projections                                       2          8   
H4 - Conférences, rencontres, lectures                          37         12   
H4 - Séances de contes                                           0          5   
Total                                                           95         38   

                                                   Nb personnes touchées  \
sll_type_public                             Total a/ Tout Type de public   
sll_type_action                                                            
H4 - Autres                                    58                    640   
H4 - Clubs de lecteurs, ateliers d'écriture    12                    130   
H4 - Concerts, projections                     10                    125   
H4 - Conférences, rencontres, lectures         48                    771   
H4 - Séances de contes                          5                      0   
Total                                         133                   1666   

                                                              
sll_type_public                             b/ Enfants Total  
sll_type_action                                               
H4 - Autres                                        399  1039  
H4 - Clubs de lecteurs, ateliers d'écriture          8   138  
H4 - Concerts, projections                         514   639  
H4 - Conférences, rencontres, lectures             449  1220  
H4 - Séances de contes                             143   143  
Total                                             1513  3179

In [139]:
ac

,Evenement,Date début action,Date fin action,Champ d'action,Type d'action,Nom action ou projet,Lieu,Espace (si dans les murs),Espace (si hors les murs et pas dans la structure partenaire,Partenariat (ne plus utiliser à terme),...,Nb adultes touchés,Nb d'accueil ou séance,Nb personnes touchées,Total personnes distinctes touchées,Volume personnes touchées,Notes,Jauge,sll_type_action,sll_type_public,Nb actions
204,NaN,2024-01-03,NaN,NaN,Jeux-vidéo,Jeux vidéo,NaN,La criée,NaN,NaN,...,NaN,NaN,60.0,60,0,NaN,NaN,H4 - Autres,NaN,1
205,NaN,2024-01-06,NaN,NaN,Jeux-vidéo,Jeux vidéo,NaN,La criée,NaN,NaN,...,NaN,NaN,55.0,55,0,NaN,NaN,H4 - Autres,NaN,1
206,NaN,2024-01-09,NaN,NaN,Lecture,Racontées des bébés,NaN,NaN,NaN,NaN,...,14.0,NaN,33.0,33,33,NaN,NaN,"H4 - Conférences, rencontres, lectures",NaN,1
207,NaN,2024-01-10,NaN,NaN,Lecture,Racontées des marmots,NaN,NaN,NaN,NaN,...,2.0,NaN,24.0,24,24,NaN,NaN,"H4 - Conférences, rencontres, lectures",NaN,1
208,NaN,2024-01-10,NaN,NaN,Lecture,Le temps des histoires,NaN,NaN,NaN,NaN,...,10.0,NaN,30.0,30,30,NaN,NaN,"H4 - Conférences, rencontres, lectures",NaN,1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
529,NaN,2024-12-20,NaN,NaN,Atelier,Radio à La Grand-Plage,NaN,Salle VDM,NaN,NaN,...,NaN,NaN,4.0,4,0,NaN,15.0,H4 - Autres,a/ Tout Type de public,1
530,NaN,2024-12-21,NaN,NaN,Atelier,Atelier de conversation en français,NaN,Salle VDM,NaN,NaN,...,NaN,NaN,7.0,7,0,NaN,10.0,H4 - Autres,a/ Tout Type de public,1
531,NaN,2024-12-21,NaN,NaN,Jeux-vidéo,Petit déjeuner jeux vidéo,NaN,La criée,NaN,NaN,...,NaN,NaN,12.0,12,0,NaN,12.0,H4 - Autres,a/ Tout Type de public,1
532,NaN,2024-12-21,NaN,NaN,Lecture,Les racontées musicales,NaN,Salle d'heure du conte,NaN,NaN,...,23.0,NaN,48.0,48,48,25 enfants et 23 adultes,40.0,"H4 - Conférences, rencontres, lectures",b/ Enfants,1


## H5 - Actions hors de l'établissement

In [140]:
aecs.groupby("Lieu")[("Nb d'accueil ou séance","Total personnes touchées")].sum()

,Nb d'accueil ou séance,Total personnes touchées
Lieu,,
Dans les murs,722.0,10412.0
Hors les murs,1963.0,1137.0
Les deux,90.0,1069.0


In [141]:
nb_actions_hors_les_murs = aecs[aecs["Lieu"]=='Hors les murs']['Total personnes touchées'].sum() + nb_actions_prets_malles_2024

In [142]:
nb_actions_hors_les_murs

9625.0

### Action hors les murs

### Portage à domicile
Le portage à domicile est effectué par M. qui transmet à F. les chiffres chaque année.

## H7 - Actions et services à destination de publics à besoins spécifiques 

In [143]:
aecs[aecs['Type de structure']=="Association"].pivot_table(index=["Nom d'équipement","Champ d'action"],
                                                   values=["Nb d'accueil ou séance","Total personnes touchées"],
                                                   margins=True,
                                                   aggfunc=sum)

,,Nb d'accueil ou séance,Total personnes touchées
Nom d'équipement,Champ d'action,,
3cm,Culture,1.0,145.0
ALEFPA,Culture,1.0,22.0
ARA,Culture,1.0,15.0
Accorderie,Culture,5.0,45.0
Amitié Partage,Culture,5.0,19.0
Armée du Salut,Culture,2.0,18.0
CIDFF (Centre d'Information sur les droits des femmes et des familles),Culture,5.0,40.0
Detournoyment,Culture,2.0,33.0
La Clef,Culture,2.0,27.0


In [144]:
liste_publics_specifiques = ["Personnes âgées (65 ans et plus)",
                             "Personnes en situation de handicap",
                             "Jeunes (18-25 ans)",
                             "Petite enfance (0-3 ans)",
                             "Personnes en recherche d'emploi",
                             "Personnes en situation d'illettrisme",
                             "Populations allophones",
                             "Populations en situation d'insertion sociale"
                            ]

In [145]:
aecs.loc[aecs["Type de public"]=="Personnes âgées (65 ans et plus)",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "a/ Personnes âgées"

aecs.loc[aecs["Type de public"]=="Personnes en situation de handicap",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "b/ Personnes en situation de handicap"

aecs.loc[aecs["Type de public"]=="Jeunes (18-25 ans)",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "c/ Jeunes"

aecs.loc[aecs["Type de public"]=="Petite enfance (0-3 ans)",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "d/ Petite enfance"

aecs.loc[aecs["Type de public"]=="Personnes en recherche d'emploi",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "e/ Personnes en recherche d'emploi"

aecs.loc[aecs["Type de public"]=="Personnes en situation d'illettrisme",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "f/ Personnes en situation d'illettrisme"

aecs.loc[aecs["Type de public"]=="Populations allophones",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "g/ Populations non-francophone"

aecs.loc[aecs["Type de public"]=="Populations en situation d'insertion sociale",
         ["H7 - Actions et services à destination de publics à besoins spécifiques"]] = "h/ Populations en situation d'insertion sociale"

In [146]:
aecs.pivot_table(index="H7 - Actions et services à destination de publics à besoins spécifiques",
                 values=['nb',"Total personnes touchées"],
                 dropna=False,
                 aggfunc={'nb':sum,"Total personnes touchées":sum}
                )[['nb','Total personnes touchées']]

,nb,Total personnes touchées
H7 - Actions et services à destination de publics à besoins spécifiques,,
a/ Personnes âgées,8,63.0
b/ Personnes en situation de handicap,1,16.0
c/ Jeunes,10,197.0
d/ Petite enfance,101,2062.0
g/ Populations non-francophone,23,348.0
h/ Populations en situation d'insertion sociale,3,27.0


# K - Patrimoine

## K2 - Fonds et acquisitions

In [147]:
query = f"""SELECT
                    i.itemnumber,
                    i.barcode,
                    i.dateaccessioned,
                    i.price,
                    i.homebranch,
                    i.holdingbranch,
                    i.location,
                    i.ccode,
                    i.itemcallnumber,
                    i.notforloan,
                    i.damaged,
                    DATE(i.damaged_on),
                    i.withdrawn,
                    DATE(i.withdrawn_on),
                    i.itemlost,
                    DATE(i.itemlost_on),
                    i.onloan,
                    i.datelastborrowed,
                    i.biblionumber,
                    b.title as titre,
                    bi.publicationyear,
                    bi.itemtype,
                    i.timestamp
                FROM koha{year}.items i
                JOIN koha{year}.biblioitems bi ON i.biblionumber = bi.biblionumber
                JOIN koha{year}.biblio b ON i.biblionumber = b.biblionumber
                WHERE i.notforloan IN (-1, -2, 0, 2, -4, 5)
                AND i.homebranch != 'MUS'"""
items = pd.read_sql(query, db_conn)

In [148]:
exemplaires = Document(df=items, db_conn=db_conn)
exemplaires.get_doc_statdb_data()
exemplaires.get_doc_es_data()

In [149]:
pat_ccode = ['PENACZZ', 'PENCVZZ', 'PENDEZZ', 'PENHPZZ', 'PENPDZZ', 'PENRSZZ', 'AAPPRLP',
             'PPAFIZZ', 'PPEFGZZ', 'PPELGZZ', 'PPEPMZZ', 'PPEPRZZ', 'PPIPIZZ', 'P17']
patrimoine_df = exemplaires.df[exemplaires.df['doc_item_collection_ccode'].isin(pat_ccode)]
patrimoine_df['sll_acq'] = 'non'
patrimoine_df.loc[patrimoine_df['doc_item_date_creation'].astype('str').str[0:4] == '2022', 'sll_acq' ] = 'oui'

In [150]:
len(patrimoine_df)

63291

### Livres imprimés

In [151]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D1 - Livres imprimés']

In [152]:
len(data)

36081

In [153]:
len(data[data['sll_acq'] == 'oui'])

440

In [154]:
data['publicationyear_'] = data['publicationyear'].str.extract(r'(^\d{4})').astype(float)
l1811 = data[data['publicationyear_'] < 1811]
len(l1811)

1054

In [155]:
len(l1811[l1811['sll_acq'] == 'oui'])

33

In [156]:
len(patrimoine_df[patrimoine_df['doc_item_collection_ccode'] == 'AAPPRLP'])

546

In [157]:
len(patrimoine_df[(patrimoine_df['doc_item_collection_ccode'] == 'AAPPRLP') & (patrimoine_df['sll_acq'] == 'oui')])

0

### Publications en séries

In [158]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D1 - Publications en série imprimées']
len(data)

2201

In [159]:
len(data[data['sll_acq'] == 'oui'])

5

### Cartes

In [160]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D3 – Documents cartographiques']
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 0
Acquisitions : 0


### Partitions

In [161]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D3 – Musique imprimée']
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 1664
Acquisitions : 90


### Documents iconographiques

In [162]:
data = patrimoine_df[patrimoine_df['sll_collection'] == 'D3 - Documents graphiques']
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 94
Acquisitions : 0


### Documents sonores

In [163]:
data = patrimoine_df[patrimoine_df['sll_collection'].isin(['D4 - Documents audiovisuels fonds adultes / Documents sonores : musique',
                                                           'D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés'])]
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 22575
Acquisitions : 161


### Images animées

In [164]:
data = patrimoine_df[patrimoine_df['sll_collection'].isin(['D4 - Documents audiovisuels fonds adultes / documents vidéo adultes',
                                                           'D4 - Documents audiovisuels fonds enfants / documents vidéo enfants'])]
print(f"Fonds : {len(data)}")
print(f"Acquisitions : {len(data[data['sll_acq'] == 'oui'])}")

Fonds : 645
Acquisitions : 0


D1 - Livres imprimés                                                                  205053
D4 - Documents audiovisuels fonds adultes / Documents sonores : musique                45233
D4 - Documents audiovisuels fonds adultes / documents vidéo adultes                    12067
D1 - Publications en série imprimées                                                    9173
D3 – Musique imprimée                                                                   3193
D4 - Documents audiovisuels fonds enfants / documents vidéo enfants                     1593
D4 - Documents audiovisuels fonds adultes / Documents sonores : livres enregistrés      1541
D4 - Documents audiovisuels fonds enfants / Documents sonores : musique                  901
D4 - Total documents multimédia sur support                                              287
D4 - Documents audiovisuels fonds enfants / Documents sonores : livres enregistrés       218
D3 – Documents cartographiques                                                           125
D3 - Documents graphiques                                                                 93
D3 - Autres documents                                                                     83